src/shared/components/plots/WaterfallPlot.spec.tsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import { assert } from 'chai';
+import { IBaseWaterfallPlotData, IWaterfallPlotProps } from './WaterfallPlot';
+import WaterfallPlot from 'shared/components/plots/WaterfallPlot';
+import { shallow } from 'enzyme';
+import labelMobxPromises from 'mobxpromise';
+
+describe('', () => {
+
+    let testData:IBaseWaterfallPlotData[] = [
+        {value: 2},
+        {value: 4}
+    ];
+    let testProps:IWaterfallPlotProps<IBaseWaterfallPlotData> = {
+        data: testData,
+        chartWidth: 600,
+        chartHeight: 400,
+        horizontal: true,
+        sortOrder: "ASC"
+    };
+    let plot:WaterfallPlot<IBaseWaterfallPlotData> = new WaterfallPlot<IBaseWaterfallPlotData>(testProps);
+    
+    beforeEach(() => {
+        
+        testProps = {
+            data: testData,
+            chartWidth: 600,
+            chartHeight: 400,
+            horizontal: true,
+            sortOrder: "ASC"
+        };
+        
+        testData = [
+            {value: 2},
+            {value: 4}
+        ];
+        
+        plot = new WaterfallPlot<IBaseWaterfallPlotData>(testProps);
+
+    });
+
+    describe('Data points', () => {
+
+        it('Datum has internally computed field', () => {
+
+            const datum = plot.internalData[0];
+            assert.isDefined(datum.order);
+
+        });
+
+        it('Datum has styling info when passed to component', () => {
+
+            testProps.fill = "test";
+            testProps.stroke = "test";
+            testProps.strokeOpacity = 1;
+            testProps.strokeWidth = 1;
+            testProps.symbol = "test";
+            testProps.labelVisibility = () => true;
+
+            plot = new WaterfallPlot<IBaseWaterfallPlotData>(testProps);
+            const datum = plot.internalData[0];
+
+            assert.isDefined(datum.fill);
+            assert.isDefined(datum.stroke);
+            assert.isDefined(datum.strokeOpacity);
+            assert.isDefined(datum.strokeWidth);
+            assert.isDefined(datum.symbol);
+            assert.isDefined(datum.labelVisibility);
+            
+        });
+
+        // it('Negative values are be log-transformed', () => {
```

Comment:
delete commented code

Reply:
No, it should be removed here since it is a left over. I have moved the formulation of log-scaling functions to the PlotsTab component (`Plotst=Tab` line 2055). Reason is that we required data type-dependent rather than plot type dependent application of data transformation types (treatment response data is log10 transformed, molecular data log2-transformed). I will move and implement the test to PlotsTab.spec.ts. Agree?

Reply:
that sounds good :+1: